src/services/job_service.py

```diff
@@ -96,25 +112,58 @@ def _get_job(self, job_uid: UUID) -> Union[Job, None]:
 
     @staticmethod
     def _get_job_entity(dmss_id: str, token: str = None):
+        """Get a document from DMSS.
+
+        The dmss_id can be on the formats:
+          - By id: PROTOCOL://DATA SOURCE/$ID.Attribute
+          - By path: PROTOCOL://DATA SOURCE/ROOT PACKAGE/SUB PACKAGE/ENTITY.Attribute
+        """
         data_source_id, job_entity_id, attribute = split_absolute_ref(dmss_id)
         return get_document_by_uid(
             reference=f"{data_source_id}/{job_entity_id}.{attribute}", token=token, depth=50, resolve_links=False
         )
 
     @staticmethod
     def _insert_reference(document_id: str, reference: dict, token: str = ""):  # nosec
+        """Insert a reference into an existing entity stored in dmss.
+
+        - **document_id**: the address to the entity we want to update. Can be on the formats:
+          - By id: PROTOCOL://DATA SOURCE/$ID.Attribute
+          - By path: PROTOCOL://DATA SOURCE/ROOT PACKAGE/SUB PACKAGE/ENTITY.Attribute
+
+        - **reference**: an entity of type 'dmss://system/SIMOS/Reference' to be inserted.
+        """
         headers = {"Access-Key": token}
+        # TODO use document update instead, the reference insert endpoint has been removed from DMSS
         req = requests.put(f"{config.DMSS_API}/api/reference/{document_id}", json=reference, headers=headers)
         req.raise_for_status()
 
         return req.json()
 
     @staticmethod
     def _update_job_entity(dmss_id: str, job_entity: dict, token: str):
+        """Update a job entity in dmss.
+
+        - **dmss_id**: the address to the job entity we want to update. Can be on the formats:
+          - By id: PROTOCOL://DATA SOURCE/$ID.Attribute
+          - By path: PROTOCOL://DATA SOURCE/ROOT PACKAGE/SUB PACKAGE/ENTITY.Attribute
+
+        - **job_entity**: the new job entity.
+        """
         return update_document_by_uid(dmss_id, {"data": job_entity}, token=token)
 
     def _get_job_handler(self, job: Job) -> JobHandlerInterface:
-        data_source_id = job.dmss_id.split("/", 1)[0]
+        """Get the job handler for a job.
+
+        Job handlers must be placed in the folders "default_job_handlers" and "job_handler_plugins" in the
```

Comment:
```suggestion
        Job handlers must be placed in the "job_handlers" folder in the
```
, right?

Reply:
ups, it is actually "in the "default_job_handlers" or "job_handler_plugins" folder in the repository src folder."